pkg/core/app/app.go

```diff
@@ -93,11 +95,15 @@ func (a *App) KeployIPv4Addr() string {
 }
 
 func (a *App) ContainerIPv4Addr() string {
-	return <-a.containerIPv4
+	a.containerIPv4Mux.RLock()         // 1. Lock for reading
```

Comment:
What happens if you continue using the channel-based approach? If it was not causing any issue, please revert .